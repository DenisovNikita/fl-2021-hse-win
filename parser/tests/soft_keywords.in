begin function main 00
begin body
    var int2 int10 = 10
end body
end function