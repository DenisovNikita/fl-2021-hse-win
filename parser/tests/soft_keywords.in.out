{
    function: {
        signature: main 0
        body: {
            bound: {
                int2 int10 = 0
            }
            bound: {
                int10 y = 15
            }
            bound: {
                string s = lolkek
            }
            only_if: {
                condition: x == 0
                body: {
                    skip: {
                    }
                    change: {
                        y = 
                        10
                    }
                }
            }
            while: {
                condition: y < 20
                body: {
                    change: {
                        y = 
                        y + 02
                    }
                    change: {
                        x = 
                        x + 11
                    }
                }
            }
            call: {
                inc 1 int10 y
            }
            if: {
                condition: y == 21
                if_body: {
                    skip: {
                    }
                }
                else_body: {
                    skip: {
                    }
                }
            }
        }
    }
    function: {
        signature: inc 1 int10 x
        body: {
            change: {
                x = 
                x + 01
            }
        }
    }
}
